### PR TITLE
cli: amend syntax and structure of cli

### LINKED ISSUE #27: cli: amend current commands
Currently we have:
```console
$ reana-client --help
Usage: reana-client [OPTIONS] COMMAND [ARGS]...

  REANA Client for interacting with REANA Server.

Options:
  -l, --loglevel [debug|info]  Sets log level
  --help                       Show this message and exit.

Commands:
  list      List all available analyses.
  ping      Health check REANA Server.
  run       Run a REANA compatible analysis using REANA...
  seed      Seed files to analysis workspace.
  validate  Validate given REANA specification file.
```

So:
* `reana-client validate` 
  -> `reana-client analysis validate`
* `reana-client list`
  -> `reana-client workflow list`
* `reana-client seed data.csv -a <workflow_uuid>` 
  -> `reana-client inputs upload data.csv --workflow <workflow_name>`
* `reana-client run` 
  -> `reana-client workflow create`
  ---> `reana-client workflow start`

Those two last have the peculiarity that, workflow create will send the spec, and start will be a simple post to `/api/analysis/:uuid` where `uuid` is the `workflow uuid` that you will get in return when doing `reana-client workflow create`.

### PR DESCRIPTION
**Pull request by hjhsalo**
* Change cli syntax and structure to use new style commands
  (see #22)
  (closes #27)

Signed-off-by: Harri Hirvonsalo <harri.hirvonsalo@cern.ch>

### CODE REVIEW

File: reana_client/cli/analyses.py [hunk 1]
```diff
@@ -19,27 +19,40 @@
...
 import click
 
 from ..config import (default_organisation, default_user,
                       reana_yaml_default_file_path)
+
 from ..utils import load_reana_spec, load_workflow_spec
 
 
+@click.group(
+    help='All interaction related analysis on REANA cloud.')
```
Comment (diegodelemos) [inline reana_client/cli/analyses.py]: `All interaction related to analyses on REANA cloud.`

File: reana_client/cli/analyses.py [hunk 2]
```diff
@@ -19,27 +19,40 @@
...
+@click.option(
+    '-f',
+    '--file',
+    type=click.Path(exists=True, resolve_path=True),
+    default=reana_yaml_default_file_path,
+    help='REANA specifications file describing the workflow and '
+         'context which REANA should execute.')
 @click.pass_context
-def validate(ctx, file):
+def cli_analysis_validate(ctx, file):
```
Comment (diegodelemos) [inline reana_client/cli/analyses.py]: Why the `cli_` prefix? it is already namespaced by the package isn't it?

Comment (diegodelemos) [reply reana_client/cli/analyses.py]: and also `analysis_` it is namespaced by the group command `analyses`

File: reana_client/cli/analyses.py (hunk 1 above)
Comment (hjhsalo) [reply reana_client/cli/analyses.py]: Fixed

File: reana_client/cli/workflow.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+class _WorkflowStatus(Enum):
+    created = 0
+    running = 1
+    finished = 2
+    failed = 3
+
+
+@click.group(
+    help='All interaction related workflows on REANA cloud.')
```
Comment (diegodelemos) [inline reana_client/cli/workflow.py]: ... related to workflows on ...

**Pull request merged**